tests/test_dunder.rs

```diff
@@ -552,3 +553,77 @@ fn getattr_doesnt_override_member() {
     py_assert!(py, inst, "inst.data == 4");
     py_assert!(py, inst, "inst.a == 8");
 }
+
+/// Wraps a Python future and yield it once.
+#[pyclass]
+struct OnceFuture {
+    future: PyObject,
+    polled: bool,
+}
+
+#[pymethods]
+impl OnceFuture {
+    #[new]
+    fn new(future: PyObject) -> Self {
+        OnceFuture {
+            future,
+            polled: false,
+        }
+    }
+}
+
+#[pyproto]
+impl PyAsyncProtocol for OnceFuture {
+    fn __await__(slf: PyRef<Self>) -> PyResult<Py<Self>> {
+        Ok(slf.into())
+    }
+}
+
+#[pyproto]
+impl PyIterProtocol for OnceFuture {
+    fn __iter__(slf: PyRef<Self>) -> PyResult<Py<Self>> {
+        Ok(slf.into())
+    }
```

Comment:
Why `__iter__` / `__next__` instead of `__aiter__` / `__anext__` ?

Also can this method just return `PyRef<Self>` ?

Reply:
Good catch :+1:

Reply:
> Why __iter__ / __next__ instead of __aiter__ / __anext__ ?

See the [document](https://docs.python.org/3/reference/datamodel.html#object.__await__).
I feel it confusing but `__await__` should return iterator.